pkg/services/apiserver/service.go

```diff
@@ -341,25 +339,31 @@ func (s *service) start(ctx context.Context) error {
 	s.options = o
 
 	delegate := server
-	var aggregatorServer *aggregatorapiserver.APIAggregator
+	var runningServer *genericapiserver.GenericAPIServer
+
 	if s.features.IsEnabledGlobally(featuremgmt.FlagKubernetesAggregator) {
-		aggregatorServer, err = s.createKubeAggregator(serverConfig, server, s.metrics)
-		if err != nil {
-			return err
+		// Run extra runners
```

Comment:
It's a bit weird that runners are only currently consumed when KubernetesAggregator is enabled.

If you go to my enterprise PR you will see that both Configure and Run functions are noop if this flag is not set.

So I would very much like to just run the runners always given another thing might want to use it. However, for Run to work as intended for the noop (flag=false case), I will need some way of making the Run for enterprise injected runner at least start the core api server. With the current signature, its not possible. Thoughts?

Reply:
this does feel like it's a bit confusing having the extra runners under the aggregator feature flag. also, if there can be more than one the order of the delegates matter 🤔